applications/audio/broadcast_configuration_tool/src/main.c

```diff
@@ -378,105 +438,431 @@ static int cmd_list(const struct shell *shell, size_t argc, char **argv)
+				}
+				ret = adv_create_and_start(shell, i);
+				if (ret) {
+					return ret;
+				}
+			}
+		}
 	}
 
-	/* Get advertising set for BIG0 */
-	ret = ext_adv_populate(0, &ext_adv_data[0], ext_adv_buf[0], ARRAY_SIZE(ext_adv_buf[0]),
-			       &ext_adv_buf_cnt);
+	return 0;
+}
+
+static int broadcaster_stop(const struct shell *shell, uint8_t big_index)
+{
+	int ret;
+
+	if (!broadcast_source_is_streaming(big_index)) {
+		return 0;
+	}
+
+	ret = broadcast_source_disable(big_index);
 	if (ret) {
+		shell_error(shell, "Failed to stop broadcaster(s) %d", ret);
 		return ret;
 	}
 
-	ret = per_adv_populate(0, &per_adv_data[0], &per_adv_buf[0], 1, &per_adv_buf_cnt);
+	ret = bt_mgmt_per_adv_stop(big_index);
 	if (ret) {
+		shell_error(shell, "Failed to stop periodic advertiser");
 		return ret;
 	}
 
-	/* Start broadcaster */
-	ret = bt_mgmt_adv_start(0, ext_adv_buf[0], ext_adv_buf_cnt, &per_adv_buf[0],
-				per_adv_buf_cnt, false);
+	ret = bt_mgmt_ext_adv_stop(big_index);
 	if (ret) {
-		shell_error(shell, "Failed to start first advertiser");
+		shell_error(shell, "Failed to stop extended advertiser");
 		return ret;
 	}
 
-	shell_print(shell, "Broadcaster started");
+	net_buf_simple_reset(&base_data);
+	net_buf_simple_reset(&base_data2);
+
+	net_buf_simple_reset(&uuid_data);
+	net_buf_simple_reset(&uuid_data2);
+
+	per_adv_buf[big_index].data_len = 0;
+	per_adv_buf[big_index].type = 0;
+
+	for (size_t j = 0; j < ARRAY_SIZE(ext_adv_buf[big_index]); j++) {
+		ext_adv_buf[big_index][j].data_len = 0;
+		ext_adv_buf[big_index][j].type = 0;
+	}
+
 	return 0;
 }
 
 static int cmd_stop(const struct shell *shell, size_t argc, char **argv)
 {
 	int ret;
 
+	if (argc == 2) {
+		size_t big_index = strtoul(argv[1], NULL, 10);
+
+		if (big_index >= CONFIG_BT_ISO_MAX_BIG) {
+			shell_error(shell, "BIG index out of range");
+			return -EINVAL;
+		}
+
+		ret = broadcaster_stop(shell, big_index);
+
+	} else {
+		for (size_t i = 0; i < CONFIG_BT_ISO_MAX_BIG; i++) {
+			if (broadcast_param[i].subgroups != NULL) {
+				ret = broadcaster_stop(shell, i);
+				if (ret) {
+					return ret;
+				}
+			}
+		}
+	}
+
+	return 0;
+}
+
+static int cmd_show(const struct shell *shell, size_t argc, char **argv)
+{
 	ARG_UNUSED(argc);
 	ARG_UNUSED(argv);
 
 	for (size_t i = 0; i < CONFIG_BT_ISO_MAX_BIG; i++) {
-		if (!broadcast_source_is_streaming(i)) {
+		if (broadcast_param[i].subgroups == NULL) {
 			continue;
 		}
 
-		ret = broadcast_source_disable(i);
-		if (ret) {
-			shell_error(shell, "Failed to stop broadcaster(s) %d", ret);
-			return ret;
+		bool streaming = broadcast_source_is_streaming(i);
+
+		shell_print(shell, "BIG %d:", i);
+		shell_print(shell, "\tStreaming: %s", (streaming ? "true" : "false"));
+
+		broadcast_config_print(shell, &broadcast_param[i]);
+	}
+
+	return 0;
+}
+
+static int cmd_preset(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc < 3) {
+		shell_error(shell, "Usage: bct preset <preset> <BIG index>");
+		return -EINVAL;
```

Comment:
Not sure what you are saying the command line usage is for this!
Do I need to type: bct preset lc3_preset_16_1_1 0

Or is it: bct c3_preset_16_1_1 0

Which actually fits the parsing.

Reply:
The way it is used is `bct preset 16_2_1 0` `16_2_1` is the preset and `0` is the BIG_index